File: piper/android/app/src/main/java/io/gopiper/piper/engine/statements/text_changeCase.kt
package io.gopiper.piper.engine.statements

import android.util.Log
import io.gopiper.piper.engine.Piper
import io.gopiper.piper.util.titlecase
import org.json.JSONObject
import java.lang.Exception

suspend fun Piper.executeTextChangeCase(statement: JSONObject, onError: (e: Exception) -> Unit): String? {
    try {
        Log.d("hello", "execute_text_changeCase")
        val pointer = statement.getJSONObject("pointer")
        val value = executeStatement(pointer["value"].toString())?.toString() ?: return null

        return when (pointer["action"].toString()) {
            "UPPERCASE" -> value.uppercase()
            "LOWERCASE" -> value.lowercase()
            "TITLECASE" -> value.titlecase()
            else -> value
        }
    } catch (e: Exception) {
        onError(e)
        return null
    }
}

File: piper/extensions/chrome/src/engine/statements/text_changeCase.js
import { stringify } from "../util"


function titleCase(str) {
    return str.replace(/\w\S*/g, function(txt){return txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase();});
}


export default async function(statement, onError) {
    try {
        let pointer = statement.pointer
        let value = await this.executeStatement(stringify(pointer.value))
        
        switch(stringify(pointer.action)) {
            case "UPPERCASE":
                return value.toUpperCase()
            case "LOWERCASE":
                return value.toLowerCase()
            case "TITLECASE":
                return titleCase(value)
            default:
                return value
        }
    } catch (e) {
        onError(e)
    }

    return null
}
